hazelcast-sql/src/main/java/com/hazelcast/sql/impl/calcite/validate/HazelcastSqlValidator.java

```diff
@@ -217,6 +223,12 @@ public void setParameterConverter(int ordinal, ParameterConverter parameterConve
         parameterConverterMap.put(ordinal, parameterConverter);
     }
 
+    public Object getArgumentAt(int index) {
```

Comment:
The feature works only because we don't cache plans for file functions. Normally we should evaluate the parameter at run time, not at validation time. I'm quite reluctant to approve this as an interim solution. I'd at least mark this method as deprecated to deter using arguments during optimization. We should move the evaluation from validation-time to run-time, this way also expressions will work as expected, without surprising feature gaps.

Reply:
> The feature works only because we don't cache plans for file functions

That's exactly why it works.
We can't move evaluation to runtime as we need file table function return type during validation.